src/components/vacation-requests-table/vacation-requests-table-toolbar/toolbar-update-status-button.tsx

```diff
@@ -0,0 +1,61 @@
+import { Check, Close } from "@mui/icons-material";
+import { Button, Typography } from "@mui/material";
+import { GridRowId } from "@mui/x-data-grid";
+import { VacationRequestStatuses } from "../../../generated/client";
+import strings from "../../../localization/strings";
+
+/**
+ * Component properties
+ */
+interface UpdateStatusButtonProps {
+  updateVacationRequestStatuses: (
+    newStatus: VacationRequestStatuses,
+    selectedRowIds: GridRowId[]
+  ) => Promise<void>;
+  approval: boolean;
+  selectedRowIds: GridRowId[];
+  setSelectedRowIds: (selectedRowIds: GridRowId[]) => void;
+}
+
+/**
+ * Status update button component
+ *
+ * @param props component properties
+ */
+const UpdateStatusButton = ({
+  updateVacationRequestStatuses,
+  approval,
+  selectedRowIds,
+  setSelectedRowIds
+}: UpdateStatusButtonProps) => {
+  /**
+   * Handle update vacation status
+   */
+  const handleUpdateVacationRequestStatus = async () => {
```

Comment:
I get a 403 when trying to approve a request, does this work as it should?

Reply:
This also happens in the old time-bank-ui when trying to change someone else's vacation requests status. It seems like the API is the culprit. Need to investigate it.